### DISCUSSION TITLE
Speculative: Disentangle `getindex` and typed-array syntax

### DISCUSSION THREAD

**Discussion by BioTurboNick** [JuliaLang/julia#53481] [RFC: features for discussion]
```
julia> d = Dict
Dict

julia> d[1]
ERROR: MethodError: Cannot `convert` an object of type
  Int64 to an object of type
  Dict

julia> d = Int
Int64

julia> d[1]
1-element Vector{Int64}:
 1
```

At first glance this is surprising - how is errantly indexing a Type trying to make a vector or do conversions?

It's happening because this is just the typed-array construction/concatenation syntax. And unfortunately, but cleverly, it overloads the `getindex` syntax and method to do this.

```
julia> Float64[1]
1-element Vector{Float64}:
 1.0
```

It might be nice, in 2.0, to consider separating the `getindex` and typed-array syntax in some way.

One available syntax could be braces (related: #8470):
```
{Float64}[1]
```

Braces are universally now used in Julia to encompass type parameters, and the typed array syntax is effectively specifying a type parameter of the constructed array. `{T}[] => Vector{T}()`

**Comment by vtjnash** [comment]
Almost a duplicate of https://github.com/JuliaLang/julia/issues/53241, except that was for `setindex!` of the same

**Reply by BioTurboNick** [reply]
That did inspire me to suggest this, but there the issue is straightforward - `setindex!` isn't ever valid for types. However, `getindex` is actually valid for types... but it probably shouldn't be because it's doing something completely different than getting at an index.
